scRNA-seq-advanced/04-differential_expression.Rmd

```diff
@@ -336,125 +346,162 @@ counts_mtx
 
-```{r}
-# deseq2 requires the coldata object as a separate input
-#  in data.frame format
+- Create a `DESeq` object
+- Normalize and log transform the counts data
+- Estimate dispersions and shrink estimates
+- Fit a negative binomial model and perform hypothesis testing using Wald statistics
+
+#### Create the `DESeq2` Object
+
+First we will create the `DESeq2` object.
+To create the object we will need the unnormalized counts matrix, the metadata associated with the samples that will help tell `DESeq2` which sample belongs to which group, and the design formula. 
+The design formula is used to indicate which columns of the metadata need to be considered in the DE comparison.
+For our experiment we are comparing gene expression between different RMS subtypes, stored in the `diagnosis_group` column of the `colData` in the pseudo-bulked SCE.
+
+```{r deseq object}
+# extract the metadata for each sample from sce as data.frame
 coldata_df <- as.data.frame(colData(tumor_myoblast_sce))
   
 # set up the deseq object 
 deseq_object <- DESeq2::DESeqDataSetFromMatrix(counts(tumor_myoblast_sce),
+                                               # metadata from coldata 
                                                colData = coldata_df,
+                                               # name of column to use for grouping samples 
                                                design = ~ diagnosis_group)
 ```
 
-Use the median of ratios method for count normalization followed by regularized log transformation.
+Before we can compare gene expression we will need to normalize the counts data.
+If you look at the pseudo-bulked SCE object you should have only seen one assay, the `counts` assay.
+This is because we took the unnormalized counts from the original SCE to create the new pseudo-bulked counts which then become the input to create the `DESeq2` object.
+We don't want to sum the normalized counts because those have each been normalized within their sample and do not account for differences between samples.
+When we run `DESeq2` on our dataset, raw counts will first be normalized using size factors to account for differences in total sample counts.
+Therefore we don't have to do any normalization on our own and we can let `DESeq2` do all the work for us.
 
-```{r}
-# estimate size factors first 
+However, we want to take a look at our data before we run the DE analysis.
+In particular, we can use principal component analysis (PCA) of our pseudo-bulked dataset as an initial test of our hypothesis that ARMS and ERMS samples have different gene expression patterns.
+If there is variation in their gene expression patterns we would expect samples from each RMS subtype to separate in PC space.
+We can evaluate this by plotting PC1 and PC2.
+
+In order to create our PCA plot, we will first need to normalize our data to account for any technical variations across samples.
+
+```{r normalize}
+# estimate size factors first
```

Comment:
It feels odd to say that we don't have to normalize since DESeq takes care of it, but then we actually do normalize here. It seems like we always have to do size factors as well?
So, let's add more context about what functions we're running for DE specifically, and what we're running for exploratory viz.